cloud/parameterstore/parameter_manager.go

```diff
@@ -0,0 +1,196 @@
+package parameterstore
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	ssmTypes "github.com/aws/aws-sdk-go-v2/service/ssm/types"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+// Parameter represents a parameter kept in Parameter Store.
+type Parameter struct {
+	// Name is the full name of the parameter, including its path. This is a
+	// unique identifier. For example, the full name could be
+	// /evergreen/path/to/my-parameter.
+	Name string `bson:"-" json:"-" yaml:"-"`
+	// Basename is the parameter's name without any hierarchical paths. For
+	// example, if the full name including the path is
+	// /evergreen/path/to/my-parameter, the basename is my-parameter.
+	Basename string `bson:"-" json:"-" yaml:"-"`
+	// Value is the parameter's plaintext value. This value should never be
+	// persisted in the DB.
+	Value string `bson:"-" json:"-" yaml:"-"`
+}
+
+// ParameterManager is an intermediate abstraction layer for interacting with
+// parameters in AWS Systems Manager Parameter Store. It supports caching to
+// optimize parameter retrieval.
+type ParameterManager struct {
+	// pathPrefix is the prefix path in the Parameter Store hierarchy. If set,
+	// all parameters should be stored under this prefix.
+	pathPrefix string
+	// cachingEnabled indicates whether parameter caching is enabled. If
+	// enabled, the cache will reduce the number of reads from Parameter Store
+	// by only fetching directly from Parameter Store if the value is missing
+	// from the cache or is stale.
+	cachingEnabled bool
+	ssmClient      SSMClient
+	db             *mongo.Database
+}
+
+// NewParameterManager creates a new ParameterManager instance.
+func NewParameterManager(pathPrefix string, cachingEnabled bool, ssmClient SSMClient, db *mongo.Database) *ParameterManager {
+	if pathPrefix != "" {
+		pathPrefix = fmt.Sprintf("/%s/", strings.TrimPrefix(strings.TrimSuffix(pathPrefix, "/"), "/"))
+	}
+	return &ParameterManager{
+		pathPrefix:     pathPrefix,
+		cachingEnabled: cachingEnabled,
+		ssmClient:      ssmClient,
+		db:             db,
+	}
+}
+
+// Put adds or updates a parameter. This returns the created parameter.
+func (pm *ParameterManager) Put(ctx context.Context, name, value string) (*Parameter, error) {
+	fullName := pm.getPrefixedName(name)
+	if _, err := pm.ssmClient.PutParameter(ctx, &ssm.PutParameterInput{
```

Comment:
Do we not need to save PutParameterOutput.Version from PutParameter for anything?

Reply:
No, it's not necessary to make the implementation correct.